engine/runtime-instrument-common/src/main/java/org/enso/interpreter/service/ExecutionService.java

```diff
@@ -175,6 +180,52 @@ public void execute(
           SourceNotFoundException,
           UnsupportedMessageException,
           UnsupportedTypeException {
+    var future =
+        executeAsync(
+            visualizationHolder,
+            module,
+            call,
+            cache,
+            methodCallsCache,
+            syncState,
+            nextExecutionItem,
+            expressionExecutionState,
+            funCallCallback,
+            onComputedCallback,
+            onCachedCallback,
+            onExecutedVisualizationCallback);
+    resultOf(future);
+  }
+
+  /**
+   * Executes a function with given arguments, represented as runtime language-level objects.
+   *
+   * @param module the module where the call is defined
+   * @param call the call metadata.
+   * @param cache the precomputed expression values.
+   * @param methodCallsCache the storage tracking the executed method calls.
+   * @param syncState the synchronization state of runtime updates.
+   * @param nextExecutionItem the next item scheduled for execution.
+   * @param expressionExecutionState the execution state for each expression.
+   * @param funCallCallback the consumer for function call events.
+   * @param onComputedCallback the consumer of the computed value events.
+   * @param onCachedCallback the consumer of the cached value events.
+   * @param onExecutedVisualizationCallback the consumer of an executed visualization result.
+   * @return future to track the status of the execution
+   */
+  private final CompletableFuture<?> executeAsync(
```

Comment:
- The change 08c21aeb6d4737f07c59301410b828e85fbffb3a illustrates possible future of _parallel execution_
- all API methods in `ExecutionService` would return `CompletableFuture`
   - right now the `executeAsync` method is `private`
   - and `public execute` method delegates to it
   - in future `execute` should be removed and `executeAsync` would be `public`
- aligned with https://github.com/enso-org/enso/issues/11429#issuecomment-2903297203 - e.g. safepoint interruptible futures
- the _language server_ schedules execution
- and obtains a `CompletableFuture` to control it and act when it is done

Reply:
Changes in https://github.com/enso-org/enso/pull/13219#pullrequestreview-2903867319 seem good. No worry, this PR will not integrate soon. I am afraid.